src/analysis/asymptotics/negligible.lean

```diff
@@ -0,0 +1,204 @@
+  (h : ∀ᶠ n in filter.at_top, ∥f n∥ ≤ ∥g n∥) : negligible f :=
+negligible_of_is_O hg $ is_O_iff.2 ⟨1, by simpa only [one_mul] using h⟩
+
+/-- It suffices to check the negligiblity condition for only sufficiently small exponents `c`.
+  See `negligible_of_is_O_fpow_le` for a version with explicit bounds -/
+lemma negligible_of_eventually_is_O
+  (h : ∀ᶠ (c : ℤ) in filter.at_bot, is_O f (λ n, (n : ℝ) ^ c) filter.at_top) :
+  negligible f :=
+begin
+  obtain ⟨C, hC⟩ := filter.eventually_at_bot.mp h,
+  intro c,
+  by_cases hc : c ≤ C,
+  { exact hC c hc },
+  { refine (hC C le_rfl).trans (is_O.of_bound 1 (filter.eventually_at_top.2 ⟨1, (λ b hb, _)⟩)),
+    simp_rw [one_mul, normed_field.norm_fpow, real.norm_coe_nat],
+    have hb : 1 ≤ (b : ℝ) := le_trans (le_of_eq nat.cast_one.symm) (nat.cast_le.2 hb),
+    exact fpow_le_of_le hb (le_of_not_le hc) }
+end
+
+lemma negligible_of_is_O_fpow_le (C : ℤ)
+  (h : ∀ c ≤ C, is_O f (λ n, (n : ℝ) ^ c) filter.at_top) :
+  negligible f :=
+negligible_of_eventually_is_O (filter.eventually_at_bot.2 ⟨C, h⟩)
+
+lemma negligible_of_is_O_fpow_lt (C : ℤ)
+  (h : ∀ c < C, is_O f (λ n, (n : ℝ) ^ c) filter.at_top) :
+  negligible f :=
+negligible_of_is_O_fpow_le C.pred
+  (λ c hc, h c (lt_of_le_of_lt hc (int.pred_self_lt C)))
+
+/-- A negligible function must tend to zero in the base ring (not just in norm) -/
+lemma tendsto_zero_of_negligible (hf : negligible f) :
+  filter.tendsto f filter.at_top (nhds 0) :=
+begin
+  refine is_O.trans_tendsto (hf (-1)) _,
+  have : (has_inv.inv : ℝ → ℝ) ∘ (coe : ℕ → ℝ) = (λ (n : ℕ), (n : ℝ) ^ (-1 : ℤ)),
+  by simp only [gpow_one, fpow_neg],
+  exact this ▸ filter.tendsto.comp (tendsto_inv_at_top_zero) (coe_nat_tendsto_at_top ℝ),
+end
+
+/-- A negligible function eventually has norm less than any positive bound -/
+lemma norm_eventually_le_of_negligible
+  (hf : negligible f) (ε : ℝ) (hε : 0 < ε) :
+  ∀ᶠ (n : ℕ) in filter.at_top, ∥f n∥ ≤ ε :=
+begin
+  obtain ⟨c, hc⟩ := is_O_iff.1 (hf (-1)),
+  have : ∀ᶠ (n : ℕ) in filter.at_top, c * ∥(n : ℝ) ^ (-1 : ℤ)∥ ≤ ε,
+  { obtain ⟨a, ha⟩ := exists_nat_ge (c * ε⁻¹),
+    refine filter.eventually_at_top.2 ⟨max a 1, λ b hb, _⟩,
+    have hb0 : 0 < (b : ℝ) := nat.cast_pos.2 (le_trans (le_max_right a 1) hb),
+    have hba : (a : ℝ) ≤ (b : ℝ) := nat.cast_le.2 (le_trans (le_max_left a 1) hb),
+    rw [fpow_neg, gpow_one, normed_field.norm_inv, real.norm_coe_nat,
+      mul_inv_le_iff hb0, mul_comm _ ε],
+    calc c ≤ ε * (a : ℝ) : (mul_inv_le_iff hε).1 ha
+      ... ≤ ε * (b : ℝ) : mul_le_mul le_rfl hba (nat.cast_nonneg a) (le_of_lt hε) },
+  exact filter.eventually.mp hc (filter.eventually.mono this (λ x hx hx', le_trans hx' hx)),
+end
+
+@[simp]
+lemma negligible_zero : negligible (function.const ℕ (0 : R)) :=
+λ c, is_O_zero _ _
+
+lemma negligible_add (hf : negligible f) (hg : negligible g) :
+  negligible (f + g) :=
+λ c, is_O.add (hf c) (hg c)
+
+lemma negligible_mul (hf : negligible f) (hg : negligible g) :
+  negligible (f * g) :=
```

Comment:
I think that you can prove
```lean
lemma negligible.mul_is_O (hf : negligible f) {c : ℤ} (hg : is_O g (λ n, (n ^ c : ℝ))) :
  negligible (f * g) :=
```
Moreover, you can use any `F : normed_group_hom E (normed_group_hom F G)` instead of multiplication.